plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/custom/util/ScriptGenerationUtil.java

```diff
@@ -295,7 +295,7 @@ public static String removeInvalidCharaters(String variableName) {
 		if (variableName.startsWith("{")) {
 			return variableName;
 		}
-		return variableName.replace(':', '_').replace(",", "_Separat0r_").replace('=', '_');
+		return variableName.replace(':', '_').replace(",", "_Separat0r_").replace('=', '_').replace("-", "_EnC0DeCHaRHyPh3n_");
```

Comment:
Can we please have a constant declared for "_EnC0DeCHaRHyPh3n_" with a proper name?